src/Diagnostics/atmos_default.jl

```diff
@@ -194,65 +241,87 @@ function compute_diagnosticsums!(
     horzavgs,
     dsums,
 )
-    zvals = CollectedDiagnostics.zvals
     th = thermo_vars(thermoQ[ijk, e])
     ha = horzavg_vars(horzavgs[Nqk * (ev - 1) + k])
     ds = diagnostic_vars(dsums[Nqk * (ev - 1) + k])
 
     u = state.ρu[1] / state.ρ
+    avg_u = ha.ρu / ha.ρ
     v = state.ρu[2] / state.ρ
+    avg_v = ha.ρv / ha.ρ
     w = state.ρu[3] / state.ρ
-    ũ = ha.ρu / ha.ρ
-    ṽ = ha.ρv / ha.ρ
-    w̃ = ha.ρw / ha.ρ
-    ẽ = ha.e_tot / ha.ρ
-    q̃_tot = ha.ρq_tot / ha.ρ
-    # TODO: temporary fix
-    if isa(atmos.moisture, EquilMoist)
-        q_tot = state.moisture.ρq_tot / state.ρ
-    end
+    avg_w = ha.ρw / ha.ρ
+
+    q_liq = th.ρq_liq / state.ρ
+    avg_q_liq = ha.ρq_liq / ha.ρ
+    q_vap = th.ρq_vap / state.ρ
+    avg_q_vap = ha.ρq_vap / ha.ρ
+    θ_liq_ice = th.ρθ_liq_ice / state.ρ
+    avg_θ_liq_ice = ha.ρθ_liq_ice / ha.ρ
+    θ_dry = th.ρθ_dry / state.ρ
+    avg_θ_dry = ha.ρθ_dry / ha.ρ
+    θ_v = th.ρθ_v
+    avg_θ_v = ha.ρθ_v / ha.ρ
+    e_int = th.ρe_int / state.ρ
+    avg_e_int = ha.ρe_int / ha.ρ
+
+    avg_e_tot = ha.ρe_tot / ha.ρ
+    avg_h_moi = ha.ρh_moi / ha.ρ
+    avg_h_tot = ha.ρh_tot / ha.ρ
+    avg_qt_sgs = ha.ρqt_sgs / ha.ρ
+    avg_ht_sgs = ha.ρht_sgs / ha.ρ
 
     # state and functions of state
-    ds.u += MH * ũ
-    ds.v += MH * ṽ
-    ds.w += MH * w̃
-    ds.e_tot += MH * ẽ
-    ds.q_tot += MH * ha.ρq_tot / ha.ρ
-    ds.q_liq += MH * ha.q_liq
-    ds.thd += MH * ha.θ_dry
-    ds.thl += MH * ha.θ_liq_ice
-    ds.thv += MH * ha.θ_v
-    ds.e_int += MH * ha.e_int
-    ds.h_m += MH * ha.h_m
-    ds.h_t += MH * ha.h_t
-    ds.qt_sgs += MH * ha.qt_sgs
-    ds.ht_sgs += MH * ha.ht_sgs
-
-    # vertical fluxes
-    ds.vert_eddy_mass_flux += MH * (w - w̃) * (state.ρ - ha.ρ)
-    ds.vert_eddy_u_flux += MH * (w - w̃) * (u - ha.ρu / ha.ρ)
-    ds.vert_eddy_v_flux += MH * (w - w̃) * (v - ha.ρv / ha.ρ)
-    ds.vert_eddy_ql_flux += MH * (w - w̃) * (th.q_liq - ha.q_liq)
-    ds.vert_eddy_qv_flux += MH * (w - w̃) * (th.q_vap - ha.q_vap)
-    ds.vert_eddy_thd_flux += MH * (w - w̃) * (th.θ_dry - ha.θ_dry)
-    ds.vert_eddy_thv_flux += MH * (w - w̃) * (th.θ_v - ha.θ_v)
-    ds.vert_eddy_thl_flux += MH * (w - w̃) * (th.θ_liq_ice - ha.θ_liq_ice)
-    # TODO: temporary fix
+    ds.rho += MH * ha.ρ
+    ds.u += MH * avg_u
+    ds.v += MH * avg_v
+    ds.w += MH * avg_w
+    ds.et += MH * avg_e_tot
+    ds.ql += MH * avg_q_liq
+    ds.thd += MH * avg_θ_dry
+    ds.thl += MH * avg_θ_liq_ice
+    ds.thv += MH * avg_θ_v
+    ds.ei += MH * avg_e_int
+    ds.hm += MH * avg_h_moi
+    ds.ht += MH * avg_h_tot
+    ds.qt_sgs += MH * avg_qt_sgs
+    ds.ht_sgs += MH * avg_ht_sgs
+
+    # covariances and variances
+    ds.cov_w_rho += MH * (w - avg_w) * (state.ρ - ha.ρ)
```

Comment:
I have talked with @tapios and we want density averaged density here (yes, it's strange). So you would calculate th.ρρ = state.ρ*state.ρ in `compute_thermo`, and avg_ρ = ha.ρρ / ha.ρ, and use avg_ρ instead of ha.ρ here, and only here.